lib/ansible/modules/cloud/misc/qemu_img.py

```diff
@@ -0,0 +1,151 @@
+    default: present
+    required: false
+notes:
+  - This module does not change the type of the image.
+'''
+
+EXAMPLES = '''
+  - description: Create a raw image of 5M.
+    code: qemu_img dest=/tmp/testimg size=5 format=raw
+
+  - description: Enlarge the image to 6M.
+    code: qemu_img dest=/tmp/testimg size=6 format=raw
+
+  - description: Remove the image
+    code: qemu_img dest=/tmp/testimg state=absent
+'''
+
+
+RETURN = '''
+# create qemu image
+present:
+  type: string
+  sample: "success"
+  returned: success
+# remove qemu image
+absent:
+  type: string
+  sample: "success"
+  returned: success
+# resize qemu image
+resize:
+  type: string
+  sample: "sucess"
+  returned: success
+'''
+
+
+import os
+from ansible.module_utils.basic import *
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            dest=dict(type='str', required=True),
+            format=dict(type='str', default='qcow2'),
+            options=dict(type='str', default='preallocation=metadata'),
+            size=dict(type='str'),
+            state=dict(type='str', choices=['absent', 'present', 'resize'], default='present'),
+        ),
+        supports_check_mode=True,
+    )
+
+    result = dict(
+        changed=False,
+        failed=False,
+    )
+
+    dest = module.params['dest']
+    img_format = module.params['format']
+    img_options = module.params['options']
+    size = module.params['size']
+
+    qemu_img = module.get_bin_path('qemu-img', True)
+
+    if module.params['state'] == 'present':
+        if not size:
+            module.fail_json(msg='No size defined, creating a disk image requires a size')
+        if not os.path.exists(dest):
+            if img_options:
+                module.run_command('%s create -f %s -o "%s" "%s" %s' % (qemu_img, img_format, img_options, dest, size), check_rc=True)
+            else:
+                module.run_command('%s create -f %s "%s" %s' % (qemu_img, img_format, dest, size), check_rc=True)
+            result['changed'] = True
+
+    if module.params['state'] == 'resize':
+        if not size:
+            module.fail_json(msg='No size defined, resizing a disk image requires a size')
+        rc, stdout, _ = module.run_command('%s info "%s"' % (qemu_img, dest), check_rc=True)
+        current_size = None
+        for line in stdout.splitlines():
+            if 'virtual size' in line:
+                ### virtual size: 5.0M (5242880 bytes)
+                current_size = int(line.split('(')[1].split()[0])
+        if not current_size:
+            module.fail_json(msg='Unable to read virtual disk size of %s'%(dest))
+        if current_size != size:
+            module.run_command('%s resize "%s" %s' % (qemu_img, dest, size), check_rc=True)
+            result.changed = True
+
+    if module.params['state'] == 'absent':
+        if os.path.exists(dest):
+            os.remove(dest)
+            result.changed = True
+
+    if module.check_mode:
```

Comment:
I am a bit scared by this. Because the module indicates it supports check-mode, but if I look at the code it seems to be doing all the stuff and only at the end appear to say there hasn't been a change. That cannot be right (it's dangerous!).